stabilizer/src/main/java/com/hazelcast/stabilizer/tests/map/PredicatesTest.java

```diff
@@ -0,0 +1,438 @@
+            return "EmployeeImpl{" +
+                    "id=" + id +
+                    ", name='" + name + '\'' +
+                    ", age=" + age +
+                    ", active=" + active +
+                    ", salary=" + salary +
+                    '}';
+        }
+    }
+
+    public static class DataSerializableEmployee extends EmployeeImpl implements DataSerializable {
+
+        public DataSerializableEmployee() {
+        }
+
+        public DataSerializableEmployee(int id) {
+            this.id = id;
+            randomizeProperties();
+        }
+
+        @Override
+        public void writeData(ObjectDataOutput objectDataOutput) throws IOException {
+            objectDataOutput.writeInt(id);
+            objectDataOutput.writeUTF(name);
+            objectDataOutput.writeInt(age);
+            objectDataOutput.writeBoolean(active);
+            objectDataOutput.writeDouble(salary);
+        }
+
+        @Override
+        public void readData(ObjectDataInput objectDataInput) throws IOException {
+            id = objectDataInput.readInt();
+            name = objectDataInput.readUTF();
+            age = objectDataInput.readInt();
+            active = objectDataInput.readBoolean();
+            salary = objectDataInput.readDouble();
+        }
+    }
+
+    public static class EmployeeDataSerializableFactory implements DataSerializableFactory{
+
+        public static final int FACTORY_ID = 3000;
+
+        @Override
+        public IdentifiedDataSerializable create(int i) {
+                return new IdentifiedDataSerializableEmployee();
+        }
+    }
+
+
+    public static class IdentifiedDataSerializableEmployee extends EmployeeImpl implements IdentifiedDataSerializable{
+
+        public IdentifiedDataSerializableEmployee() {
+        }
+
+        public IdentifiedDataSerializableEmployee(int id) {
+            this.id = id;
+            randomizeProperties();
+        }
+
+        @Override
+        public int getFactoryId() {
+            return EmployeeDataSerializableFactory.FACTORY_ID;
+        }
+
+        @Override
+        public int getId() {
+            return 0;
+        }
+
+        @Override
+        public void writeData(ObjectDataOutput objectDataOutput) throws IOException {
+            objectDataOutput.writeInt(id);
+            objectDataOutput.writeUTF(name);
+            objectDataOutput.writeInt(age);
+            objectDataOutput.writeBoolean(active);
+            objectDataOutput.writeDouble(salary);
+        }
+
+        @Override
+        public void readData(ObjectDataInput objectDataInput) throws IOException {
+            id = objectDataInput.readInt();
+            name = objectDataInput.readUTF();
+            age = objectDataInput.readInt();
+            active = objectDataInput.readBoolean();
+            salary = objectDataInput.readDouble();
+        }
+    }
+
+    public static class SerializableEmployee extends EmployeeImpl implements Serializable {
```

Comment:
I don't know if this trick is going to work because EmployeeImpl is the super class and is not serializable and afaik its fields will not be serialized/deserialized.